server/src/internalClusterTest/java/org/elasticsearch/snapshots/SharedClusterSnapshotRestoreIT.java

```diff
@@ -1207,6 +1208,13 @@ public void testSnapshotStatus() throws Exception {
         assertThat(getResponse.getSnapshots().size(), equalTo(1));
         SnapshotInfo snapshotInfo = getResponse.getSnapshots().get(0);
         assertThat(snapshotInfo.state(), equalTo(SnapshotState.IN_PROGRESS));
+        snapshotStatus = client.admin().cluster().prepareSnapshotStatus().execute().actionGet().getSnapshots().get(0);
+        assertThat(snapshotInfo.totalShards(), equalTo(snapshotStatus.getIndices().get("test-idx").getShardsStats().getTotalShards()));
+        assertThat(
+            snapshotInfo.successfulShards(),
+            lessThanOrEqualTo(snapshotStatus.getIndices().get("test-idx").getShardsStats().getDoneShards())
```

Comment:
I had to relax the condition to `lessThanOrEqualTo` from `equalTo`, because the `getSnapshot`  and `snapshotStatus` API calls happen with a delay, so we can't guarantee that `successfulShards` and `doneShards` are actually the same